src/clj/athenaeum/handlers/user.clj

```diff
@@ -35,49 +44,47 @@
        (walk/keywordize-keys)
        (reformat-payload)))
 
-(defn- find-or-create-user
-  [payload]
-  (if-let [user (db/with-transaction [tx @db/datasource]
-                  (user/fetch-by-google-id tx (:google-id payload)))]
-    user
-    (db/with-transaction [tx @db/datasource]
-      (user/create tx payload))))
-
 (defn- set-session-id-cookie
   [response session-id]
   (response/set-cookie response
                        :session-id
                        session-id
                        {:same-site :strict
-                        :max-age   3600
+                        :max-age   (get @config/config :session-ttl)
                         :path      "/api"}))
 
 (defn login
   [{:keys [headers]}]
-  (if-let [payload (some-> headers
-                           :id-token
-                           verify-id-token
-                           get-payload)]
-    (if (= (:domain payload) "nilenso.com")
-      (-> (response/response {:message "login success"})
-          (set-session-id-cookie (-> payload
-                                     find-or-create-user
-                                     :id
-                                     session/create)))
-      (-> (response/response {:message "invalid domain"})
-          (response/status 401)))
-    (response/bad-request {:message "id token verification failed"})))
+  (if-let [id-token (-> headers
```

Comment:
You should use `some->` here, otherwise `str/split` might throw in case of a `nil`.